packages/react/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,33 +1,292 @@
+      const listElementArray = Array.from(listElement.children) as HTMLElement[]
+      const widths = listElementArray.map(child => child.offsetWidth)
+      setChildArrayWidths(widths)
+      setRootItemWidth(listElementArray[0].offsetWidth)
+    }
+  }, [childArray.length])
+
+  // Measure actual menu button width when it exists
+  useEffect(() => {
+    if (menuButtonRef.current) {
+      const measuredWidth = menuButtonRef.current.offsetWidth
+      if (measuredWidth > 0) {
+        setMenuButtonWidth(measuredWidth)
+      }
+    }
+  }, [menuItems.length]) // Re-measure when menu button appears/disappears
+
+  const calculateOverflow = useCallback(
+    (availableWidth: number) => {
+      const MENU_BUTTON_WIDTH = menuButtonWidth // Use measured width with fallback
+
+      const calculateVisibleItemsWidth = (w: number[]) => {
+        const widths = w.reduce((sum, width) => sum + width + 16, 0)
+        return !effectiveHideRoot ? rootItemWidth + widths : widths
+      }
+
+      let currentVisibleItems = [...childArray]
+      let currentVisibleItemWidths = [...childArrayWidths]
+      let currentMenuItems: React.ReactElement[] = []
+      let currentMenuItemsWidths: number[] = []
+      let eHideRoot = effectiveHideRoot
+
+      if (availableWidth > 0 && currentVisibleItemWidths.length > 0) {
+        let visibleItemsWidthTotal = calculateVisibleItemsWidth(currentVisibleItemWidths)
+
+        // Add menu button width if we have hidden items
+        if (currentMenuItems.length > 0) {
+          visibleItemsWidthTotal += MENU_BUTTON_WIDTH
+        }
+        while (
+          overflow === 'menu' &&
+          (visibleItemsWidthTotal > availableWidth || currentVisibleItems.length > MIN_VISIBLE_ITEMS)
+        ) {
+          // Remove the last visible item
+          const itemToHide = currentVisibleItems.slice(0)[0]
+          const itemToHideWidth = currentVisibleItemWidths.slice(0)[0]
+          currentMenuItems = [...currentMenuItems, itemToHide]
+          currentMenuItemsWidths = [...currentMenuItemsWidths, itemToHideWidth]
+          currentVisibleItems = [...currentVisibleItems.slice(1)]
+          currentVisibleItemWidths = [...currentVisibleItemWidths.slice(1)]
+
+          visibleItemsWidthTotal = calculateVisibleItemsWidth(currentVisibleItemWidths)
+
+          // Add menu button width
+          if (currentMenuItems.length > 0) {
+            visibleItemsWidthTotal += MENU_BUTTON_WIDTH
+          }
+
+          // If hideRoot is false but we still don't fit with root + menu + leaf,
+          // fallback to hideRoot=true behavior (menu + leaf only)
+          if (!hideRoot && currentVisibleItems.length === 1 && visibleItemsWidthTotal > availableWidth) {
+            eHideRoot = true
+            break
+          } else {
+            eHideRoot = hideRoot
+          }
+        }
+      }
+      return {
+        visibleItems: [...currentVisibleItems],
+        menuItems: [...currentMenuItems],
```

Comment:
One idea with the spread:

```suggestion
        visibleItems: currentVisibleItems,
        menuItems: currentMenuItems,
```

Let me know if these spread suggestions aren't applicable here btw!